plugins/scaffolder/src/components/TemplateEditorPage/DryRunContext.tsx

```diff
@@ -122,6 +124,22 @@ export function DryRunProvider(props: DryRunProviderProps) {
     });
   }, []);
 
+  const downloadResult = useCallback((id: number) => {
+    setState(prevState => {
+      const index = prevState.results.findIndex(r => r.id === id);
+      if (index === -1) {
+        return prevState;
+      }
+      const result = prevState.results[index];
+      console.log('Result', result.id);
+      createZipDownload(result.directoryContents, 'result_' + result.id)
+      return {
```

Comment:
do we need this `setState`? I see you are setting `results` and `selectedResult` to exactly the same values 🤔

Reply:
moved to DryRunResultsList component